src/actions/breakpoints.js

```diff
@@ -82,6 +128,47 @@ export function enableBreakpoint(location: Location) {
   };
 }
 
+export function syncBreakpoint(sourceId, pendingBreakpoint) {
+  return ({ dispatch, getState, client, sourceMaps }: ThunkArgs) => {
+    const { line, sourceUrl, column } = pendingBreakpoint.location;
+    const location = { sourceId, sourceUrl, line, column };
+    const source = getSource(getState(), sourceId);
+
+    let promise = syncClientBreakpoint(
+      source,
+      client,
+      sourceMaps,
+      pendingBreakpoint,
+      location
+    );
+
+    const breakpoint = _createBreakpoint(location, pendingBreakpoint);
+
+    // If the breakpoint is already disabled, we need to communicate
+    // with the server, but we also have to remove the breakpoint immediately.
+    if (pendingBreakpoint.disabled) {
```

Comment:
Good point here. Originally, my thinking is that we need to verify our breakpoints with the server, and the only way to do that (at the time at least) was to add a breakpoint to the server. If its disabled we need to then remove it. Otherwise we have the weird behavior of disabled breakpoints in places that don't make sense.

Better solution: change the client code so we can get a breakpoint location. I haven't had a chance yet to try to make this better but it feels like it should be possible